### PR TITLE
[WIP] Initial implementation of Modal on Android

### PR DESCRIPTION
**Pull request by satya164**
The `<Modal />` component has the same API as on iOS - http://facebook.github.io/react-native/docs/modal.html#content

The `animated` prop is not implemented yet.

### CODE REVIEW

File: ReactAndroid/src/main/java/com/facebook/react/views/modalhost/ReactModalHostView.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package com.facebook.react.views.modalhost;
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/modalhost/ReactModalHostView.java]: nit: add a license header please

File: ReactAndroid/src/main/java/com/facebook/react/views/modalhost/ReactModalHostManager.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/modalhost/ReactModalHostManager.java]: nit: add the BSD license header please

File: ReactAndroid/src/main/java/com/facebook/react/views/modalhost/DismissEvent.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.facebook.react.views.modalhost;
```
Comment (mkonicek) [inline ReactAndroid/src/main/java/com/facebook/react/views/modalhost/DismissEvent.java]: nit: add the BSC license header please

**Comment by satya164** [comment]
@mkonicek Done :)

**Comment by mkonicek** [comment]
Wow that was fast! I assume I'll have to update some internal Buck files in order to merge this, let's see :)

@facebook-github-bot shipit

**Comment by satya164** [comment]
Import failed? @mkonicek

**Comment by mkonicek** [comment]
Yup need to update BUCK files.

**Comment by satya164** [comment]
Resolved conflicts.

**Comment by Ehesp** [comment]
Nice work :)

**Comment by mkonicek** [comment]
Cool, will update the Buck files and merge this. I'm doing an internal hackathon today with Shawn Wilsher from the Buck team trying to get OSS RN to build with Buck. Will move the release to Monday.

@facebook-github-bot shipit

**Comment by mkonicek** [comment]
Does the Modal work well for you? I've added `ModalExample` to `UIExplorerList.android.js`, here's a screen recording: https://youtu.be/nRe_Clx4Y-k

**Comment by satya164** [comment]
@mkonicek I had only tried closing the modal with back button. Will investigate what's happening. Thanks.

**Comment by satya164** [comment]
So, I've been trying to fix the `<Modal />` the whole night. The issue is that the view inside the modal doesn't receive touch. I think this is because the `createViewInstance` method returns a `ReactModalHostView` which is initialized and configured to receive touches and stuff, but it's not really used anywhere. A different `ReactViewGroup` instance is created which is added to the dialog, and it's not configured to respond to touches.

**Comment by satya164** [comment]
cc @AaaChiuuu

**Comment by mkonicek** [comment]
Thanks for looking into it and sorry about that, I assumed it worked out of the box!

**Comment by satya164** [comment]
@mkonicek Me too. Let's get it fixed. It's gonna save me a bunch of really ugly code :D

**Comment by mpretty-homepass** [comment]
@satya164 @mkonicek has there been any movement on this? I'm about to start adding in some platform checks to avoid attempting to display UI modally but would love to avoid having to do that if this PR is going to drop some time soon.

**Comment by satya164** [comment]
@mpretty-homepass Nope. There are some technical difficulties we need to handle before this can be merged. It might take some time.

**Comment by mpretty-homepass** [comment]
@satya164 Ok I'll set Android up to push all modals for the time being, thanks for the update.

**Comment by satya164** [comment]
`<Modal />` was open sourced in db7a1543608f45a0ad39b80564f22f5eca135822

**Pull request closed without merging**